x/bank/keeper/keeper_test.go

```diff
@@ -0,0 +1,305 @@
+				accBalance: 800,
+				fromAddr:   suite.address[0],
+				toAddr:     suite.address[1],
+			},
+			errArgs{
+				shouldPass: true,
+				contains:   "",
+			},
+		},
+		{"Sender(1) inValid Coins: SendAmt = 0",
+			args{
+				sendAmt:    0,
+				accBalance: 1000,
+				fromAddr:   suite.address[0],
+				toAddr:     suite.address[1],
+			},
+			errArgs{
+				shouldPass: false,
+				contains:   "",
+			},
+		},
+		{"Sender(1) inValid: SendAmt > Total Amount",
+			args{
+				sendAmt:    11000,
+				accBalance: 1000,
+				fromAddr:   suite.address[0],
+				toAddr:     suite.address[1],
+			},
+			errArgs{
+				shouldPass: false,
+				contains:   "",
+			},
+		},
+	}
+	for _, tc := range tests {
+		suite.Run(tc.name, func() {
+			suite.SetupTest()
+			err := suite.keeper.SendCoins(suite.ctx, tc.args.fromAddr, tc.args.toAddr, sdk.Coins{sdk.NewInt64Coin("uctk", tc.args.sendAmt)})
+			balance := suite.keeper.GetAllBalances(suite.ctx, tc.args.fromAddr)
+			if tc.errArgs.shouldPass {
+				suite.Require().NoError(err, tc.name)
+				suite.Require().NotEqual(tc.args.sendAmt, balance)
+				suite.Require().Equal(sdk.Coins{sdk.NewInt64Coin("uctk", tc.args.accBalance)}, balance)
+			} else {
+				suite.Require().Error(err, tc.name)
+				suite.Require().True(strings.Contains(err.Error(), tc.errArgs.contains))
+			}
+		})
+	}
+}
+
+func (suite *KeeperTestSuite) TestInputOutputCoins() {
+	type args struct {
+		Addr1        sdk.AccAddress
+		Addr2        sdk.AccAddress
+		Addr3        sdk.AccAddress
+		amount       int64
+		addr1Balance int64
+		addr2Balance int64
+		addr3Balance int64
+	}
+
+	type errArgs struct {
+		shouldPass bool
+		contains   string
+	}
+
+	tests := []struct {
+		name    string
+		args    args
+		errArgs errArgs
+	}{
+		{"Sender(1) isValid: Sufficient Amount",
+			args{
+				amount:       200,
+				Addr1:        suite.address[0],
+				Addr2:        suite.address[1],
+				Addr3:        suite.address[2],
+				addr1Balance: 600,
+				addr2Balance: 1200,
+				addr3Balance: 1200,
+			},
+			errArgs{
+				shouldPass: true,
+				contains:   "",
+			},
+		},
+		{"Sender(1) inValid: InSufficient Amount",
+			args{
+				amount:       5000,
+				Addr1:        suite.address[0],
+				Addr2:        suite.address[1],
+				Addr3:        suite.address[2],
+				addr1Balance: 1000,
+				addr2Balance: 1000,
+				addr3Balance: 1000,
+			},
+			errArgs{
+				shouldPass: false,
+				contains:   "",
+			},
+		},
+	}
+	for _, tc := range tests {
+		suite.Run(tc.name, func() {
+			suite.SetupTest()
+			inputs := []bankTypes.Input{
+				{Address: tc.args.Addr1.String(), Coins: sdk.Coins{sdk.NewInt64Coin("uctk", tc.args.amount)}},
+				{Address: tc.args.Addr1.String(), Coins: sdk.Coins{sdk.NewInt64Coin("uctk", tc.args.amount)}},
+			}
+			outputs := []bankTypes.Output{
+				{Address: tc.args.Addr2.String(), Coins: sdk.Coins{sdk.NewInt64Coin("uctk", tc.args.amount)}},
+				{Address: tc.args.Addr3.String(), Coins: sdk.Coins{sdk.NewInt64Coin("uctk", tc.args.amount)}},
```

Comment:
never use hard coded "uctk" in the rest of the codebase

Reply:
done